### PR TITLE
Fixes issue with multiple noop causing noop_2

### PR DESCRIPTION
**Pull request by kingbuzzman**
Adds a couple of things:

* Adds `UniqueVariableName.function` to keep track of the function, along with the name that is going to be used in the squash migration. Fixes an issue where the names of functions looked like `func_2_2_2_2...`
* Writer knows now whether or not to put the function in the file or not. (if its inside `site-packages` it doesnt add the function) (one of the issues with `Migration.noop`)
* Adds tests

### CODE REVIEW

File: django_squash/management/commands/lib/autodetector.py [hunk 1]
```diff
@@ -54,6 +54,34 @@ def from_migration(cls, migration):
...
+
+        if inspect.ismethod(func) or inspect.signature(func).parameters.get('self') is not None:
+            raise ValueError('func cannot be part of an instance')
+
+        name = original_name = func.__qualname__
+        already_accounted = func in self.functions
+        if already_accounted:
+            return self.functions[func]
+
+        for i, _ in enumerate(itertools.count(), 2):
```
Comment (philfriesen) [inline django_squash/management/commands/lib/autodetector.py:74]: This loop is mysterious. Please add a comment explaining in plain English what it does.

File: django_squash/management/commands/lib/autodetector.py [hunk 2]
```diff
@@ -133,9 +161,11 @@ def add_non_elidables(self, original, loader, changes):
                 for operation in all_custom_operations(migration.operations, unique_names):
                     if isinstance(operation, migration_module.RunPython):
                         operation.code = copy_func(operation.code)
+                        operation.code.__original_module__ = operation.code.__module__
```
Comment (philfriesen) [inline django_squash/management/commands/lib/autodetector.py:164]: You can ignore this suggestion, but when adding our properties to someone else's object, you want to name the property in a way that clearly identifies that it is ours and the creators of that object will never use our property name. Instead of naming it dunder style, which the original authors might use in the future, I would prefix props we add to their objects with "squasher_". Example: "squasher_original_module".

File: tests/app/tests/migrations/run_python_noop/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+
+def same_name(apps, schema_editor):
+    # original function
+    return
+
+
+def same_name_2(apps, schema_editor):
+    # original function 2
+    return
```
Comment (philfriesen) [inline tests/app/tests/migrations/run_python_noop/0001_initial.py:17]: I think you need to prove that you can handle any number of suffixes. To do that, I would make this test use "same_name_3" and "same_name_5". I think you need to prove that you can handle multiple similar suffixes, both consecutive ones (2 & 3) as well as gaps (3 & 5).

Do you also handle suffixes so that the same migration could also have "same_nameA", "same_name_A",  "same_nameB", "same_name_B", "same_name2"?

**Review by philfriesen — CHANGES_REQUESTED** [review]

File: django_squash/management/commands/lib/autodetector.py (hunk 1 above)
Comment (kingbuzzman) [reply django_squash/management/commands/lib/autodetector.py:74]: LOL. Its an endless loop. that starts at 2 :P -- will do

File: tests/app/tests/migrations/run_python_noop/0001_initial.py (hunk 3 above)
Comment (kingbuzzman) [reply tests/app/tests/migrations/run_python_noop/0001_initial.py:17]: I prove this in tests_utils.py::TestUtils.test_unique_function_names -- didn't see the need to do here also.

File: django_squash/management/commands/lib/autodetector.py (hunk 2 above)
Comment (kingbuzzman) [reply django_squash/management/commands/lib/autodetector.py:164]: This is the kind of thing typehint/pyright will absolutely hate me for, but i can't figure doing any other way. `code` is the function object itself. Unless python in the future starts to use `__original_module__` we _should_ be ok.

File: django_squash/management/commands/lib/autodetector.py (hunk 1 above)
Comment (kingbuzzman) [reply django_squash/management/commands/lib/autodetector.py:75]: Done.
